src/gatsby-node.js

```diff
@@ -90,17 +90,19 @@ exports.onCreatePage = async ({ page, actions }, pluginOptions) => {
       },
     }
   }
-
-  const newPage = generatePage(false, defaultLanguage)
+  
+  const pageLanguage = page.context.language
+  const newPage = generatePage(!!pageLanguage, pageLanguage || defaultLanguage)
```

Comment:
if `pageLanguage` is the default language, this would wrongly create a route for the page.

This is my suggestion:
```suggestion
const route = !!pageLanguage && pageLanguage !== defaultLanguage
const newPage = generatePage(route, pageLanguage || defaultLanguage)
```

Summary:
* `pageLang === undefined` -> route is `false`
* `pageLang === defaultLang` -> route is `false`
* `pageLang !=== defaultLang` -> route is `true`

Reply:
Thanks! would you apply the same logic to line 98 below? i.e. using `if(!route)` instead of `if (!pageLanguage)` ?

Reply:
No, you should actually leave to depend on `pageLanguage`.

A more elegant way imo, is to break out of execution when `pageLanguage` has a value, like I did [here](https://github.com/wiziple/gatsby-plugin-intl/pull/169/files).